### ISSUE TITLE
[Feature]: Display subcrate version information

### ISSUE THREAD

**Issue by platten** [enarx/enarx#1449]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Following a discussion with @haraldh on 22-02-2022: For debugging purposes, it would be useful to see which versions (w/ preferably git hashes) of the subcrates were used for a particular enarx build and displays that information when running `enarx --version`.

### Acceptance Criteria

- [ ] Store information around explicitly specified subcrates (sallyport, xsave, etc) in the built binary
- [ ] Display that information when running `enarx --version`

**Comment by stephssmiths** [comment]
If nobody has taken this issue yet - I would love to try implement it.

**Comment by stephssmiths** [comment]
Current commit https://github.com/stephssmiths/enarx-fork-debug/commit/97505d0c611a9b30b3582f07353b564ea830bbd1

**Comment by bstrie** [comment]
@stephssmiths It looks like there is a project out there that seeks to achieve a similar result: https://github.com/Shnatsel/rust-audit . Care to take a look at it and see how it differs from what you've got so far?

**Comment by bstrie** [comment]
There's also a proposal here for Cargo to have first-class support for this: https://github.com/rust-lang/rfcs/pull/2801

**Comment by haraldh** [comment]
yeah, I read that article about rust flaws, too, but forgot to mention it.
